sources/osgUtil/Intersector.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var intersectionEnums = require('osgUtil/intersectionEnums');
+
+var Intersector = function() {
+    this._intersections = [];
+    this._primitiveMask = intersectionEnums.ALL_PRIMITIVES;
+    this._intersectionLimit = intersectionEnums.NO_LIMIT;
+};
+
+Intersector.prototype = {
+    reset: function() {
+        this._intersections.length = 0;
+    },
+
+    enter: function(node) {
+        if (this.reachedLimit()) return false;
+        return !node.isCullingActive() || this.intersectBoundingSphere(node.getBound());
+    },
+
+    getIntersections: function() {
+        return this._intersections;
+    },
+
+    setIntersectionLimit: function(limit) {
+        this._intersectionLimit = limit;
+    },
+
+    getIntersectionLimit: function() {
+        return this._intersectionLimit;
+    },
+
+    setPrimitiveMask: function(primitiveMask) {
+        this._primitiveMask = primitiveMask;
+    },
+
+    getPrimitiveMask: function() {
+        return this._primitiveMask;
+    },
+
+    reachedLimit: function() {
+        return (
+            this._intersectionLimit === intersectionEnums.LIMIT_ONE &&
+            this._intersections.length > 0
+        );
+    },
+
+    setCurrentTransformation: function(/*matrix*/) {},
+    intersectBoundingSphere: function(/*bsphere*/) {},
```

Comment:
I disagree on that point, if the "API" is wrong (name/function) then it should be fixed.

Also not every functions in osgjs should be considered as "API" I think.
Because of javascript we can't differentiate virtual/private/public, but there are many functions that are simply internal/helpers, etc...

Reply:
Yes, and that's why I agree on this change :)